### PR TITLE
Review improvements

### PR DESCRIPTION
**Pull request by MarekMacko**
- presenter injection,
- data classes in separate files,
- travis integration,
- dagger network module,
- caching response,

### CODE REVIEW

File: app/src/main/java/com/marekmacko/kotlinapp/ForecastListAdapter.kt [hunk 1]
```diff
@@ -9,9 +9,9 @@ import com.marekmacko.kotlinapp.data.WeeklyForecast
 import com.marekmacko.kotlinapp.util.loadFromUrl
 import kotlinx.android.synthetic.main.item_forecast.view.*
 
-class ForecastAdapter(private val weeklyForecast: WeeklyForecast,
-                      private val itemClick: (DailyForecast) -> Unit)
-    : RecyclerView.Adapter<ForecastAdapter.ViewHolder>() {
+class ForecastListAdapter(private var weeklyForecast: WeeklyForecast?,
```
Comment (sswierczek) [inline app/src/main/java/com/marekmacko/kotlinapp/ForecastListAdapter.kt]: Could be null? Maybe consider using non-null type

File: app/src/main/java/com/marekmacko/kotlinapp/ForecastListAdapter.kt [hunk 2]
```diff
@@ -20,15 +20,20 @@ class ForecastAdapter(private val weeklyForecast: WeeklyForecast,
     }
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) =
-            holder.bindDailyForecast(weeklyForecast[position])
+            holder.bindDailyForecast(weeklyForecast!![position])
 
-    override fun getItemCount(): Int = weeklyForecast.size
+    override fun getItemCount(): Int = weeklyForecast?.size?: 0
+
+    fun setItemsAndNotify(weeklyForecast: WeeklyForecast) {
```
Comment (sswierczek) [inline app/src/main/java/com/marekmacko/kotlinapp/ForecastListAdapter.kt]: Could be just named setItems

File: app/src/main/java/com/marekmacko/kotlinapp/ForecastListAdapter.kt [hunk 3]
```diff
@@ -20,15 +20,20 @@ class ForecastAdapter(private val weeklyForecast: WeeklyForecast,
...
+        this.weeklyForecast = weeklyForecast
+        notifyDataSetChanged()
+    }
 
     inner class ViewHolder(view: View, private val itemClick: (DailyForecast) -> Unit)
         : RecyclerView.ViewHolder(view) {
 
         fun bindDailyForecast(dailyForecast: DailyForecast) = with(dailyForecast) {
-            val weather = weather[0] // TODO: API always return one element
+            val weather = weather[0]
```
Comment (sswierczek) [inline app/src/main/java/com/marekmacko/kotlinapp/ForecastListAdapter.kt]: What if array is empty or null?

File: app/src/main/java/com/marekmacko/kotlinapp/api/NetworkModule.kt [hunk 4]
```diff
@@ -1,13 +1,60 @@
...
+import retrofit2.converter.gson.GsonConverterFactory
 import javax.inject.Singleton
 
+
 @Module
-internal class NetworkModule {
+internal class NetworkModule(private val context: Context) {
+
+    companion object {
+        private const val CACHE_SIZE_MB: Long = 5
```
Comment (sswierczek) [inline app/src/main/java/com/marekmacko/kotlinapp/api/NetworkModule.kt:18]: Later you could consider some Config class for this

File: app/src/main/java/com/marekmacko/kotlinapp/data/DailyForecast.kt [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+
+data class DailyForecast(private val dt: Long, val temp: Temperature,
+                         val pressure: Float, val humidity: Int,
+                         val weather: List<Weather>, val speed: Float,
+                         val deg: Int, val clouds: Int,
+                         val rain: Float) : Serializable {
+
+    companion object {
+        private val dateFormatter by lazy {
```
Comment (sswierczek) [inline app/src/main/java/com/marekmacko/kotlinapp/data/DailyForecast.kt:15]: Nice

File: app/src/main/java/com/marekmacko/kotlinapp/ui/WeeklyForecastFragment.kt [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+import com.marekmacko.kotlinapp.mvp.WeatherMvp
+import com.marekmacko.kotlinapp.mvp.WeatherPresenter
+import com.marekmacko.kotlinapp.mvp.WeatherPresenterModule
+import com.marekmacko.kotlinapp.repository.WeatherRepository
+import kotlinx.android.synthetic.main.fragment_weekly_forecast.*
+import org.jetbrains.anko.toast
+import javax.inject.Inject
+
+
+class WeeklyForecastFragment : Fragment(), WeatherMvp.View {
```
Comment (sswierczek) [inline app/src/main/java/com/marekmacko/kotlinapp/ui/WeeklyForecastFragment.kt:23]: You could cleanup methods order a little bit

File: app/src/main/java/com/marekmacko/kotlinapp/util/Utils.kt [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package com.marekmacko.kotlinapp.util
+
+import android.content.Context
+import android.net.ConnectivityManager
+
+
+class Utils {
```
Comment (sswierczek) [inline app/src/main/java/com/marekmacko/kotlinapp/util/Utils.kt:7]: Utils are always hard to mock, guess in test you would like to mock this, just create regular object and inject via DI

**Comment by MarekMacko** [comment]
Config class and Utils will be fixed with tests branch

**Pull request merged**